src/main/java/com/minecolonies/coremod/colony/permissions/Permissions.java

```diff
@@ -680,7 +664,7 @@ public boolean setPlayerRank(final UUID id, final Rank rank, final World world)
         {
             player.setRank(rank);
 
-            if (rank == Rank.OFFICER || rank == Rank.OWNER)
+            if (rank == ranks.get(OFFICER_RANK_ID) || rank == ranks.get(OWNER_RANK_ID))
```

Comment:
Probably the flag I mentioned above should be sth like "colony manager" so we can also use this here

Reply:
still missing